zcash_proofs/src/circuit/sapling.rs

```diff
@@ -412,17 +439,65 @@ impl<'a, E: JubjubEngine> Circuit<E> for Spend<'a, E> {
 
 impl<'a, E: JubjubEngine> Circuit<E> for Output<'a, E> {
     fn synthesize<CS: ConstraintSystem<E>>(self, cs: &mut CS) -> Result<(), SynthesisError> {
+        use itertools::multizip;
         // Let's start to construct our note, which contains
         // value (big endian)
+        // asset_generator || value || g_d || pk_d
         let mut note_contents = vec![];
 
-        // Expose the value commitment and place the value
-        // in the note.
-        note_contents.extend(expose_value_commitment(
+        let mut asset_generator_preimage = Vec::with_capacity(256);
+
+        for (i, bit) in self.asset_identifier.iter().enumerate() { 
+            let cs = &mut cs.namespace(|| format!("witness asset type bit {}", i));
+
+            let asset_identifier_preimage_bit = boolean::Boolean::from(boolean::AllocatedBit::alloc(
+                cs.namespace(|| "asset type bit"),
+                *bit,
+            )?);
+
+            // Push this boolean for nullifier computation later
```

Comment:
What does this comment mean? We just need to compute the asset generator, right?